File: 802G - Fake News (easy).py
s = input()
target = "heidi"
j = 0
s_len = len(s)
for i in range(0, s_len):
    if s[i] == target[j]:
        j += 1
        if j == len(target):
            break
if (j == len(target)):
    print("YES")
else:
    print("NO")

File: 802G - Fake News (easy).cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{

    const string t("heidi");
    string s;
    cin >> s;
    bool fake(false);
    long ind(0);
    for(long p = 0; p < s.size(); p++)
    {
        if(s[p] == t[ind])
        {
            ++ind;
        }
        if(ind >= t.size())
        {
            fake = true;
            break;
        }
    }

    cout << (fake ? "YES" : "NO") << endl;

    return 0;
}
